plugins/router/template/plugin_test.go

```diff
@@ -346,7 +346,7 @@ func TestHandleRoute(t *testing.T) {
 	templatePlugin := newDefaultTemplatePlugin(router, true)
 	// TODO: move tests that rely on unique hosts to pkg/router/controller and remove them from
 	// here
-	plugin := controller.NewUniqueHost(templatePlugin, controller.HostForRoute)
+	plugin := controller.NewUniqueHost(templatePlugin, controller.HostForRoute, controller.LogRejections)
```

Comment:
Might be good to add a test using a test rejection controller that verifies the duplicate route added in this test is ignored.